File: backend/src/main/java/net/zorphy/backend/site/catan/dto/game/GameState.java
package net.zorphy.backend.site.catan.dto.game;

import net.zorphy.backend.site.all.base.GameStateBase;
import net.zorphy.backend.site.catan.dto.DicePair;
import net.zorphy.backend.site.catan.dto.DiceRoll;

import java.time.Instant;
import java.time.LocalDateTime;
import java.util.List;

public record GameState(
        Instant startTime,
        GameConfig gameConfig,
        int currentPlayerTurn,
        int currentShipTurn,
        List<DicePair> classicCards,
        List<Character> eventCards,
        List<DiceRoll> diceRolls
) implements GameStateBase {
}

File: frontend/src/app/sites/catan/dto/game/GameState.ts
import {DiceRoll} from "../DiceRoll";
import {GameConfig} from "./GameConfig";
import {DicePair} from "../DicePair";
import {GameStateBase} from "../../../all/dto/GameStateBase";

export interface GameState extends GameStateBase {
    gameConfig: GameConfig,
    currentPlayerTurn: number,
    currentShipTurn: number,
    classicCards: DicePair[],
    eventCards: string[],
    diceRolls: DiceRoll[]
}